Automattic/wp-calypso
# Skip upgrade step when plan is already selected

Related to #93300 

## Proposed Changes

* Handle cases coming to the migration flow with a plan already defined.
* Remove step `import-or-migrate-with-upgrade` which wasn't used.

### Alternative solution

I started implementing it in a way that the Upgrade Plan would be responsible for submitting automatically when the plan was already defined, but after thinking more, I decided to handle it directly in the flow, where it seems to make more sense.

## Why are these changes being made?


* We will have a case where the users will come from a landing page already selecting a plan.

## Testing Instructions



* Navigate to `/setup/migration/create-site?platform=wordpress&plan=business`, and make sure a site is created and you are redirected directly to the checkout.
* Navigate to `/setup/migration/create-site?platform=wordpress&plan=unknownplan` and make sure you are redirected to the upgrade plan step.
* Navigate through the flow `/setup/migration`, and make sure it continues working properly.

## Pre-merge Checklist
